File: python/casemap.py
#!/usr/bin/env python

import sys, time

def benchmark_once(f):
    start = time.time()
    f()
    end = time.time()
    return end - start

def benchmark(f):
    runs = 100
    total = 0.0
    for i in range(runs):
        total += benchmark_once(f)
    return total / runs

def casemap(f):
    s = open(f).read().decode('utf-8')
    s.upper()

for f in sys.argv[1:]:
    t = benchmark(lambda: casemap(f))
    print('{0}: {1}'.format(f, t))

File: ruby/casemap.rb
#!/usr/bin/env ruby

def benchmark_once
  start = Time.now
  yield
  stop = Time.now
  stop - start
end

def benchmark(&b)
  runs = 100
  total = 0.0
  runs.times do
    total += benchmark_once(&b)
  end
  total / runs
end

def casemap(f)
  File.open(f, 'r:utf-8') do |file|
    file.read.upcase
  end
end

ARGV.each do |f|
  t = benchmark { casemap(f) }
  puts "#{f}: #{t}"
end
